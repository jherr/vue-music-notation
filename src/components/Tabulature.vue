<template>
  <div>
    <TabStart
      :stringCount="stringCount"
      :tuning="tuning"
    />
    <Tab
      :chord="chord"
      v-for="(chord, index) in chords"
      :tabIndex="index"
      :key="index"
    />
    <TabEnd
      :stringCount="stringCount"
    />
  </div>
</template>

<script>
import TabStart from './base/TabStart';
import TabEnd from './base/TabEnd';
import Tab from './base/Tab';

export default {
  props: {
    stringCount: {
      default: 6,
    },
    tuning: {
      default: () => [ 'E', 'A', 'D', 'G', 'B', 'E' ],
    },
    chords: {
      default: () => [],
    },
  },
  components: {
    TabStart,
    TabEnd,
    Tab,
  },
}
</script>
