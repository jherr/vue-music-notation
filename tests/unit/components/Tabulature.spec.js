import { mount } from '@vue/test-utils';
import Tabulature from '@/components/Tabulature.vue';

describe('Tabulature', () => {
  it('renders default', () => {
    const wrapper = mount(Tabulature, {
      propsData: {
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders', () => {
    const wrapper = mount(Tabulature, {
      propsData: {
        chords: [ [ 0, 2, 2, 1, 0, 0 ], [ -1, -1, -1, 1, 2, 1 ] ],
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
