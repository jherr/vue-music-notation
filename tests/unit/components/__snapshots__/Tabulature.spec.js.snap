// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabulature renders 1`] = `
<div>
  <svg
    height="105"
    style="display: inline;"
    width="20"
  >
    <line
      style="stroke-width: 1; stroke: black;"
      x1="15"
      x2="15"
      y1="97.5"
      y2="22.5"
    />
     
    <line
      style="stroke-width: 1; stroke: black;"
      x1="15"
      x2="20"
      y1="97.5"
      y2="97.5"
    />
    <line
      style="stroke-width: 1; stroke: black;"
      x1="15"
      x2="20"
      y1="82.5"
      y2="82.5"
    />
    <line
      style="stroke-width: 1; stroke: black;"
      x1="15"
      x2="20"
      y1="67.5"
      y2="67.5"
    />
    <line
      style="stroke-width: 1; stroke: black;"
      x1="15"
      x2="20"
      y1="52.5"
      y2="52.5"
    />
    <line
      style="stroke-width: 1; stroke: black;"
      x1="15"
      x2="20"
      y1="37.5"
      y2="37.5"
    />
    <line
      style="stroke-width: 1; stroke: black;"
      x1="15"
      x2="20"
      y1="22.5"
      y2="22.5"
    />
     
    <text
      style="font-size: 12px; font-family: Arial, Verdana, Sans-Serif;"
      x="0"
      y="101.25"
    >
      
    E
  
    </text>
    <text
      style="font-size: 12px; font-family: Arial, Verdana, Sans-Serif;"
      x="0"
      y="86.25"
    >
      
    A
  
    </text>
    <text
      style="font-size: 12px; font-family: Arial, Verdana, Sans-Serif;"
      x="0"
      y="71.25"
    >
      
    D
  
    </text>
    <text
      style="font-size: 12px; font-family: Arial, Verdana, Sans-Serif;"
      x="0"
      y="56.25"
    >
      
    G
  
    </text>
    <text
      style="font-size: 12px; font-family: Arial, Verdana, Sans-Serif;"
      x="0"
      y="41.25"
    >
      
    B
  
    </text>
    <text
      style="font-size: 12px; font-family: Arial, Verdana, Sans-Serif;"
      x="0"
      y="26.25"
    >
      
    E
  
    </text>
  </svg>
   
  <svg
    height="105"
    style="display: inline;"
    width="20"
  >
    <line
      style="stroke-width: 1; stroke: black;"
      x1="0"
      x2="20"
      y1="97.5"
      y2="97.5"
    />
    <line
      style="stroke-width: 1; stroke: black;"
      x1="0"
      x2="20"
      y1="82.5"
      y2="82.5"
    />
    <line
      style="stroke-width: 1; stroke: black;"
      x1="0"
      x2="20"
      y1="67.5"
      y2="67.5"
    />
    <line
      style="stroke-width: 1; stroke: black;"
      x1="0"
      x2="20"
      y1="52.5"
      y2="52.5"
    />
    <line
      style="stroke-width: 1; stroke: black;"
      x1="0"
      x2="20"
      y1="37.5"
      y2="37.5"
    />
    <line
      style="stroke-width: 1; stroke: black;"
      x1="0"
      x2="20"
      y1="22.5"
      y2="22.5"
    />
     
    <line
      style="stroke-width: 1; stroke: white;"
      x1="3"
      x2="18"
      y1="97.5"
      y2="97.5"
    />
    <line
      style="stroke-width: 1; stroke: white;"
      x1="3"
      x2="18"
      y1="82.5"
      y2="82.5"
    />
    <line
      style="stroke-width: 1; stroke: white;"
      x1="3"
      x2="18"
      y1="67.5"
      y2="67.5"
    />
    <line
      style="stroke-width: 1; stroke: white;"
      x1="3"
      x2="18"
      y1="52.5"
      y2="52.5"
    />
    <line
      style="stroke-width: 1; stroke: white;"
      x1="3"
      x2="18"
      y1="37.5"
      y2="37.5"
    />
    <line
      style="stroke-width: 1; stroke: white;"
      x1="3"
      x2="18"
      y1="22.5"
      y2="22.5"
    />
     
    <text
      class="number"
      style="font-size: 12px;"
      x="16"
      y="101.78571428571429"
    >
      
    0
  
    </text>
    <text
      class="number"
      style="font-size: 12px;"
      x="16"
      y="86.78571428571429"
    >
      
    2
  
    </text>
    <text
      class="number"
      style="font-size: 12px;"
      x="16"
      y="71.78571428571429"
    >
      
    2
  
    </text>
    <text
      class="number"
      style="font-size: 12px;"
      x="16"
      y="56.785714285714285"
    >
      
    1
  
    </text>
    <text
      class="number"
      style="font-size: 12px;"
      x="16"
      y="41.785714285714285"
    >
      
    0
  
    </text>
    <text
      class="number"
      style="font-size: 12px;"
      x="16"
      y="26.785714285714285"
    >
      
    0
  
    </text>
  </svg>
  <svg
    height="105"
    style="display: inline;"
    width="20"
  >
    <line
      style="stroke-width: 1; stroke: black;"
      x1="0"
      x2="20"
      y1="97.5"
      y2="97.5"
    />
    <line
      style="stroke-width: 1; stroke: black;"
      x1="0"
      x2="20"
      y1="82.5"
      y2="82.5"
    />
    <line
      style="stroke-width: 1; stroke: black;"
      x1="0"
      x2="20"
      y1="67.5"
      y2="67.5"
    />
    <line
      style="stroke-width: 1; stroke: black;"
      x1="0"
      x2="20"
      y1="52.5"
      y2="52.5"
    />
    <line
      style="stroke-width: 1; stroke: black;"
      x1="0"
      x2="20"
      y1="37.5"
      y2="37.5"
    />
    <line
      style="stroke-width: 1; stroke: black;"
      x1="0"
      x2="20"
      y1="22.5"
      y2="22.5"
    />
     
    <line
      style="stroke-width: 1; stroke: white;"
      x1="3"
      x2="18"
      y1="52.5"
      y2="52.5"
    />
    <line
      style="stroke-width: 1; stroke: white;"
      x1="3"
      x2="18"
      y1="37.5"
      y2="37.5"
    />
    <line
      style="stroke-width: 1; stroke: white;"
      x1="3"
      x2="18"
      y1="22.5"
      y2="22.5"
    />
     
    <text
      class="number"
      style="font-size: 12px;"
      x="16"
      y="56.785714285714285"
    >
      
    1
  
    </text>
    <text
      class="number"
      style="font-size: 12px;"
      x="16"
      y="41.785714285714285"
    >
      
    2
  
    </text>
    <text
      class="number"
      style="font-size: 12px;"
      x="16"
      y="26.785714285714285"
    >
      
    1
  
    </text>
  </svg>
   
  <svg
    height="105"
    style="display: inline;"
    width="2"
  >
    <line
      style="stroke-width: 1; stroke: black;"
      x1="1"
      x2="1"
      y1="22.5"
      y2="97.5"
    />
  </svg>
</div>
`;

exports[`Tabulature renders default 1`] = `
<div>
  <svg
    height="105"
    style="display: inline;"
    width="20"
  >
    <line
      style="stroke-width: 1; stroke: black;"
      x1="15"
      x2="15"
      y1="97.5"
      y2="22.5"
    />
     
    <line
      style="stroke-width: 1; stroke: black;"
      x1="15"
      x2="20"
      y1="97.5"
      y2="97.5"
    />
    <line
      style="stroke-width: 1; stroke: black;"
      x1="15"
      x2="20"
      y1="82.5"
      y2="82.5"
    />
    <line
      style="stroke-width: 1; stroke: black;"
      x1="15"
      x2="20"
      y1="67.5"
      y2="67.5"
    />
    <line
      style="stroke-width: 1; stroke: black;"
      x1="15"
      x2="20"
      y1="52.5"
      y2="52.5"
    />
    <line
      style="stroke-width: 1; stroke: black;"
      x1="15"
      x2="20"
      y1="37.5"
      y2="37.5"
    />
    <line
      style="stroke-width: 1; stroke: black;"
      x1="15"
      x2="20"
      y1="22.5"
      y2="22.5"
    />
     
    <text
      style="font-size: 12px; font-family: Arial, Verdana, Sans-Serif;"
      x="0"
      y="101.25"
    >
      
    E
  
    </text>
    <text
      style="font-size: 12px; font-family: Arial, Verdana, Sans-Serif;"
      x="0"
      y="86.25"
    >
      
    A
  
    </text>
    <text
      style="font-size: 12px; font-family: Arial, Verdana, Sans-Serif;"
      x="0"
      y="71.25"
    >
      
    D
  
    </text>
    <text
      style="font-size: 12px; font-family: Arial, Verdana, Sans-Serif;"
      x="0"
      y="56.25"
    >
      
    G
  
    </text>
    <text
      style="font-size: 12px; font-family: Arial, Verdana, Sans-Serif;"
      x="0"
      y="41.25"
    >
      
    B
  
    </text>
    <text
      style="font-size: 12px; font-family: Arial, Verdana, Sans-Serif;"
      x="0"
      y="26.25"
    >
      
    E
  
    </text>
  </svg>
    
  <svg
    height="105"
    style="display: inline;"
    width="2"
  >
    <line
      style="stroke-width: 1; stroke: black;"
      x1="1"
      x2="1"
      y1="22.5"
      y2="97.5"
    />
  </svg>
</div>
`;
